src/bootstrap/setup.rs

```diff
@@ -111,21 +111,28 @@ pub fn setup(src_path: &Path, profile: Profile) {
     let build = TargetSelection::from_user(&env!("BUILD_TRIPLE"));
     let stage_path = ["build", build.rustc_target_arg(), "stage1"].join("/");
 
-    let created = fs::create_dir(&stage_path[..]).is_ok();
+    let rustup_installed = Command::new("rustup")
+        .arg("--version")
+        .stdout(std::process::Stdio::null())
+        .output()
+        .is_ok();
```

Comment:
This needs to check `status()`, remember.

Reply:
🤦